src/SynthesiserRelation.cpp

```diff
@@ -1163,7 +1163,14 @@ void SynthesiserEqrelRelation::generateTypeStruct(std::ostream& out) {
     // lowerUpperRange methods, one for each of the 4 possible search patterns
     size_t arity = 2;
     for (int i = 1; i < 4; i++) {
-        SearchSignature s(arity, i);
+        SearchSignature s(arity);
+        if (i & 1) {
+            s.set(0, AttributeConstraint::Equal);
+        }
+        if (i & 2) {
+            s.set(1, AttributeConstraint::Equal);
+        }
```

Comment:
I had to stare at this for 15+ seconds. Maybe too clever?

Reply:
Fixed